pkg/evetestkit/utils.go

```diff
@@ -0,0 +1,487 @@
+func (node *EveNode) SSHExec(appName, command string) (string, error) {
+	appConfig := node.getAppConfig(appName)
+	if appConfig == nil {
+		return "", fmt.Errorf("app %s not found, make sure to deploy app/vm with WithSSH option", appName)
+	}
+
+	host := fmt.Sprintf("%s:%s", node.ip, appConfig.sshPort)
+
+	config := &ssh.ClientConfig{
+		User: appConfig.sshUser,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(appConfig.sshPass),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         30 * time.Second,
+	}
+	client, err := ssh.Dial("tcp", host, config)
+	if err != nil {
+		return "", fmt.Errorf("failed to dial: %s", err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return "", fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	output, err := session.CombinedOutput(command)
+	if err != nil {
+		return "", fmt.Errorf("failed to run command: %s", err)
+	}
+
+	return string(output), nil
+}
+
+// SCPCopy copies a file from the local machine to the app VM running on the EVE node.
+func (node *EveNode) SCPCopy(appName, localFile, remoteFile string) error {
+	appConfig := node.getAppConfig(appName)
+	if appConfig == nil {
+		return fmt.Errorf("app %s not found, make sure to deploy app/vm with WithSSH option", appName)
+	}
+
+	host := fmt.Sprintf("%s:%s", node.ip, appConfig.sshPort)
+
+	config := &ssh.ClientConfig{
+		User: appConfig.sshUser,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(appConfig.sshPass),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         30 * time.Second,
+	}
+	client, err := ssh.Dial("tcp", host, config)
+	if err != nil {
+		return fmt.Errorf("failed to dial: %s", err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	err = scp.CopyPath(localFile, remoteFile, session)
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// WithSSH is an option that sets the SSH configuration for the app running on
+// the EVE node, this should be use with DeployVM function.
+func WithSSH(user, pass, port string) VMOption {
+	return func(n *EveNode) {
+		// slices are ordered, so the last element is the one
+		// we're going to set things for.
+		n.apps[len(n.apps)-1].sshUser = user
+		n.apps[len(n.apps)-1].sshPass = pass
+		n.apps[len(n.apps)-1].sshPort = port
+	}
+}
+
+// DeployVm deploys a VM on the EVE node
+func (node *EveNode) DeployVM(imageURL string, pc openevec.PodConfig, options ...VMOption) error {
+	node.apps = append(node.apps, appInstanceConfig{name: pc.Name})
+	for _, option := range options {
+		option(node)
+	}
+
+	return node.controller.PodDeploy(imageURL, pc, node.cfg)
+}
+
+// IsTpmEnabled checks if EVE node is running with (SW)TPM enabled
+func (node *EveNode) IsTpmEnabled() bool {
+	return node.cfg.Eve.TPM
+}
+
+func (node *EveNode) setEveIP() error {
```

Comment:
Shouldn't this be `getEveIP` or `discoverEveIP` or `retrieveEveIP` ... (since we are not really changing the Node IP address)